main/main.go

```diff
@@ -86,11 +92,11 @@ func main() {
 	}
 
 	// Initialize metrics gathered through prometheus
-	gatherer, registerer, err := initMetrics()
+	gatherer, registerer, err := initializeMetrics()
 	if err != nil {
-		logger.Fatal("failed to set up prometheus metrics",
+		logger.Fatal("Failed to set up prometheus metrics",
 			zap.Error(err))
-		panic(err)
+		return
```

Comment:
`return` after a `logger.Fatal` is very strange. I think we should either mimic the behaviour of the most common `logger.Fatal` - `panic`, or make this an error log.

Reply:
Yup, agreed. I went ahead and replaced all the "if error then log and return" with panics, with the exception of the final "Relayer exiting" log since it's the last statment of the `main` function.